test/structured_metadata_spec.js

```diff
@@ -0,0 +1,356 @@
+const expect = require("expect.js");
+const Q = require('q');
+const cloudinary = require("../cloudinary");
+const helper = require("./spechelper");
+
+const TEST_ID = helper.TEST_ID;
+const TEST_TAG = helper.TEST_TAG;
```

Comment:
There are many tests missing or incomplete. For example, add test for Test creating an integer, enum, date, set metadata filed. Please align tests to match PHP implementation- cloudinary_php/pull/198